src/zenml/model/model_version.py

```diff
@@ -170,28 +171,47 @@ def load_artifact(self, name: str, version: Optional[str] = None) -> Any:
 
         artifact = self.get_artifact(name=name, version=version)
 
-        if not artifact:
+        if not artifact or not isinstance(artifact, ArtifactResponse):
```

Comment:
Old check is redundant now

```suggestion
        if not isinstance(artifact, ArtifactResponse):
```